service/api_patch.go

```diff
@@ -109,8 +111,46 @@ func (as *APIServer) submitPatch(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	patchID := mgobson.NewObjectId()
+	author := dbUser.Id
+	grip.Debug(message.Fields{
+		"message":       "testing user",
+		"github_author": data.GithubAuthor,
+	})
+	if data.GithubAuthor != "" {
+		opts := gimlet.PermissionOpts{
+			Resource:      pref.Id,
+			ResourceType:  evergreen.ProjectResourceType,
+			Permission:    evergreen.PermissionPatches,
+			RequiredLevel: evergreen.PatchSubmitAdmin.Value,
+		}
+		if !dbUser.HasPermission(opts) {
+			as.LoggedError(w, r, http.StatusUnauthorized, errors.New("user is not authorized to patch on behalf of other users"))
+			return
+		}
+		specifiedUser, err := user.FindByGithubName(data.GithubAuthor)
+		if err != nil {
+			as.LoggedError(w, r, http.StatusInternalServerError, errors.Wrap(err, "error looking for specified author"))
+		}
+		if specifiedUser != nil {
+			grip.Info(message.Fields{
+				"message":    "overriding patch author as specified by the submitter",
+				"submitter":  dbUser.Id,
+				"new_author": data.GithubAuthor,
+				"patch_id":   patchID,
+			})
+			author = specifiedUser.Id
+		}
+		grip.DebugWhen(specifiedUser == nil, message.Fields{
+			"message":         "github user not found",
```

Comment:
Should this be a LoggedError?

Reply:
I put this here as helpful logging for us in the future, whereas I think LoggedError should only be used if we're failing the command (it's not worth failing here bc it's not the inputters fault that we can't find the github user, but the Github users fault for not including this in their user settings)